crates/tabby/src/services/health.rs

```diff
@@ -18,6 +23,8 @@ pub struct HealthState {
     arch: String,
     cpu_info: String,
     cpu_count: usize,
+    gpu_devices: Vec<String>,
+    #[deprecated(note = "Use the more generic gpu_devices instead")]
```

Comment:
```suggestion
```

Let's don't deprecate the field. There're downstream usage of this field relying on it to detect nvidia card.

Reply:
That's what the device type is for though, having 2 ways of getting the information is pretty unclean and will inevitably lead to issues. The original field wasn't even correct. And just because it's deprecated doesn't mean it's going away anytime soon. And do you want to maintain fields for every single HW vendor? It's 2 now, maybe 3 soon and the way things go it'll be 10s in the future probably, so having to maintain vendor specific fields for each won't make sense.

Keep the cuda devices for compatibility (hence it being deprecated, not removed) and let's move to a more generic approach. If that implementation doesn't suit you, you could also think about changing to `gpu_devices` list to containing a device struct, that has like a name, serial number, id and type or so, that would probably be the cleanest one.

Reply:
> The original field wasn't even correct. 

Could you provide further details on why the original field was incorrect?

The rationale for not deprecating it is straightforward: we currently have downstream compoent that relies on this specific field and is not likely to be modified in the immediate future. Therefore, I propose the following options:

> I'm fine to either:
> adding rocm device to the list directly (Since it uses a cuda ish api)
> or adding a new field gpu_devices, while keeping cuda_devices as is

Maintaining support for the cuda_devices field is crucial for these downstream components, and it is imperative that we preserve it.